### PR TITLE
Verification steps refactor

### PR DESCRIPTION
**Pull request by epeach**
Refactor the name and location of SetupStep to ValidateStep in apps/src/templates to make it more reusable outside of the maker kit.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Thoughts on `ValidateStep` vs. `ValidationStep`? Not sure how strongly I feel, but I think slight preference for the latter.

**Comment by epeach** [comment]
Re: Validate vs. Validation. Validation seems grammatically better, but Validate is shorter. I'm pretty ambivalent on it.

**Comment by islemaster** [comment]
Not sure I would have thought about it, but now that @Bjvanminnen brings it up I strongly prefer `ValidationStep` because I prefer noun-names for our UI components, where `ValidateStep` sounds like a verb.

I'd prefer we add new reusable UI components to [apps/src/lib/ui](https://github.com/code-dot-org/code-dot-org/tree/staging/apps/src/lib/ui) per [this old proposal](https://github.com/code-dot-org/code-dot-org/blob/44ca7dc61ca0774ee8345b678012fe5cd39e14b7/apps/docs/refactor.md#directory-hierarchy) from @pcardune that I still think is a good idea :grin:

File: apps/src/templates/EligibilityChecklist.jsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+      acceptedPledge: false
+    },
+    errors: {
+      invalidEmail: false
+    }
+  };
+
+  // Saves the teaching-year choice to trigger next step of actions
+  handleSubmit = () => {
+    this.setState({ statusYear: (this.state.yearChoice ? Status.SUCCEEDED : Status.FAILED )});
```
Comment (islemaster) [inline apps/src/templates/EligibilityChecklist.jsx]: Nit: Unbalanced whitespace.

File: apps/src/templates/EligibilityChecklist.jsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    });
+  }
+
+  handleNotFoundChange = (field, event) => {
+    this.setState({
+      submission: {
+        ...this.state.submission,
+        [field]: event.target.value
+      }
+    }, this.checkShowFollowUp);
```
Comment (islemaster) [inline apps/src/templates/EligibilityChecklist.jsx]: `checkShowFollowUp` doesn't seem to exist anywhere else in this file.

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  state = {
+    signature: "",
+    activeButton: false,
+  };
+
+  verifyResponse = () => {
+    if (this.check1.checked &&
+        this.check2.checked &&
+        this.check3.checked &&
+        this.state.signature !== ""){
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: Nit: I'd require some non-whitespace content in the signature to pass this check.  Easy way to do that:
```js
/\S/.test(this.state.signature)
```

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      this.setState({activeButton: true});
+    } else { // if someone unchecks their responses
+      this.setState({activeButton: false});
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: I'd flip this method's implementation inside-out so you only have one setState call:
```js
verifyResponse = () => {
  this.setState({
    activeButton: this.check1.checked
      && this.check2.checked
      && this.check3.checked
      && this.state.signature !== ""
  });
};
```

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  }
+
+  setSignature = (event) => {
+    this.setState({signature: event.target.value});
+    this.verifyResponse();
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: This won't work consistently - `setState` is an asynchronous operation, but `verifyResponse` depends on its result.  You can instead pass `verifyResponse` as a callback, like this:
```js
setSignature = (event) => {
  this.setState({signature: event.target.value}, this.verifyResponse);
};
```

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  render() {
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        uncloseable
+        isOpen = {true}
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: Redundant: Just say `isOpen`.

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        style={{padding:20}}
+      >
+        <h2>{i18n.getCode()}</h2>
+        <div style={styles.subtitle} >{i18n.verifyStatementsforCode()}</div>
+        <br/>
+        <form>
+          <label style={styles.boxStyle}>
+            <input
+              type="checkbox"
+              ref={input => {this.check1 = input;}}
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: Style nit: Braces are unnecessary
```jsx
ref={input => this.check1 = input}
```

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+              onChange={this.verifyResponse}
+            />
+            {i18n.verifySingleCode()}
+          </label>
+          <br/>
+          <label>
+            <div><b>{i18n.verifySignature()}</b> {i18n.typeName()}</div>
+            <input
+              value=""
+              style={{width: 98 + '%', height: 40}}
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: Why not `width: '98%'`?

File: apps/src/templates/EligibilityConfirmDialog.jsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+              onChange={this.setSignature}
+            />
+          </label>
+        </form>
+        <div>{i18n.contactSupport()}</div>
+        <DialogFooter>
+          <Button
+            text={i18n.dialogCancel()}
+            onClick={() => {}}
```
Comment (islemaster) [inline apps/src/templates/EligibilityConfirmDialog.jsx]: You'll probably want to take submit and cancel callbacks as props to this component - might as well add those now.

**Comment by epeach** [comment]
@islemaster @Bjvanminnen PTAL

**Pull request merged**